### PR TITLE
feat: add disabled property to grid (#3445) (CP: 14)

### PR DESCRIPTION
**Pull request by vursen**
## Description

The PR back-ports the grid disabled property feature to V14:

- https://github.com/vaadin/web-components/pull/3445

Fixes https://github.com/vaadin/web-components/issues/2170

## Type of change

- [x] Feature

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs/latest/guide/contributing/overview/
- [x] I have added a description following the guideline.
- [x] The issue is created in the corresponding repository and I have referenced it.
- [x] I have added tests to ensure my change is effective and works as intended.
- [x] New and existing tests are passing locally with my change.
- [x] I have performed self-review and corrected misspellings.

### CODE REVIEW

File: theme/lumo/vaadin-grid-styles.html [hunk 1]
```diff
@@ -29,6 +29,10 @@
         --_lumo-grid-selected-row-color: var(--lumo-primary-color-10pct);
       }
 
+      :host([disabled]) {
+        opacity: 0.7;
```
Comment (sissbruecker) [inline theme/lumo/vaadin-grid-styles.html:33]: The opacity doesn't show up when testing on iOS 13. Probably not super important though, as interaction with the grid is correctly disabled.

**Review by sissbruecker — APPROVED** [review]

File: test/visual/test.js [hunk 2]
```diff
@@ -77,6 +77,18 @@ gemini.suite('vaadin-grid', (rootSuite) => {
       });
     });
 
+    gemini.suite(`disabled-${theme}`, (suite) => {
+      suite
+        .setUrl(`disabled.html?theme=${theme}`)
+        .setCaptureElements('.capture-block')
+        .capture('disabled', {}, (actions, find) => {
```
Comment (web-padawan) [inline test/visual/test.js:84]: nit: no need to set `disabled` dynamically, you can add it as attribute to the web component.

File: test/disabled.html [hunk 3]
```diff
@@ -0,0 +1,85 @@
+<!doctype html>
+
+<html>
+<head>
+  <meta charset="UTF-8">
+  <title>iron-list test</title>
```
Comment (web-padawan) [inline test/disabled.html:6]: nit: the title is wrong.

**Review by web-padawan — APPROVED** [review]

File: test/visual/test.js (hunk 2 above)
Comment (vursen) [reply test/visual/test.js]: Thanks, done.

File: test/disabled.html (hunk 3 above)
Comment (vursen) [reply test/disabled.html]: Done. Also opened a chore PR fixing the title of other test pages: https://github.com/vaadin/vaadin-grid/pull/2212

File: theme/lumo/vaadin-grid-styles.html (hunk 1 above)
Comment (sissbruecker) [reply theme/lumo/vaadin-grid-styles.html:33]: I tested with Safari through Browserstack on an iPhone 8 with iOS 13. It doesn't say which version in particular.

Comment (vursen) [reply theme/lumo/vaadin-grid-styles.html:33]: I guess I know where the problem comes from. I disabled the `animation` property for the case when the grid is disabled. Could you check to see if this fixes the issue?

Comment (sissbruecker) [reply theme/lumo/vaadin-grid-styles.html:33]: Yep, it seems to work now 👍

**Pull request merged**